pkg/apis/core/validation/validation.go

```diff
@@ -5515,6 +5515,10 @@ func ValidatePodResize(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		allErrs = append(allErrs, field.Invalid(specPath, newPod.Status.QOSClass, "Pod QOS Class may not change as a result of resizing"))
 	}
 
+	if !isPodResizeRequestSupported(*oldPod) {
```

Comment:
@ndixita you have similar logic (different reason, same result) - PTAL and see if we can join them, whoever merges last :)